### PR TITLE
fix(TDI-47879): Removed destination property from tExtractJSONFields

### PR DESCRIPTION
**Pull request by dochkas-talend**
Removed property and added migration task to remove this property from old jobs

### CODE REVIEW

File: main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/RemoveDestinationPropertyExtractJSONFieldsMigrationTask.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                if (destination != null) {
+                    nodeType.getElementParameter().remove(destination);
+                }
+            }
+        };
+
+        IComponentFilter componentFilter = new NameComponentFilter(componentNameToAffect);
+
+        try {
+            ModifyComponentsAction.searchAndModify(item, processType,
```
Comment (ozhelezniak-talend) [inline main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/RemoveDestinationPropertyExtractJSONFieldsMigrationTask.java:47]: You should use the return value to check if smth was modified.

File: main/plugins/org.talend.repository/plugin.xml [hunk 2]
```diff
@@ -3732,7 +3732,15 @@
               id="org.talend.repository.model.migration.EncryptClientSecretMicrosoftCRMMigrationTask"
               name="EncryptClientSecretMicrosoftCRMMigrationTask"
               version="8.0.1">
-
+        </projecttask>
+		<projecttask
+              beforeLogon="true"
+              breaks="8.0.1"
```
Comment (ozhelezniak-talend) [inline main/plugins/org.talend.repository/plugin.xml:3738]: Not sure, maybe we need to use 8.0.0 here. Need to ask in tup-tdi.

**Review by ozhelezniak-talend — CHANGES_REQUESTED** [review]
migration task should be modified. 
If the Migration task doesn't change anything it should return nothing to do.

File: main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/RemoveDestinationPropertyExtractJSONFieldsMigrationTask.java (hunk 1 above)
Comment (dochkas-talend) [reply main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/RemoveDestinationPropertyExtractJSONFieldsMigrationTask.java:47]: Ok, then I should also update my migration task from TDI-47539

Comment (ozhelezniak-talend) [reply main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/RemoveDestinationPropertyExtractJSONFieldsMigrationTask.java:47]: in the scope of another work item

File: main/plugins/org.talend.repository/plugin.xml (hunk 2 above)
Comment (ozhelezniak-talend) [reply main/plugins/org.talend.repository/plugin.xml:3738]: Could you align tags?

File: main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/EncryptClientSecretMicrosoftCRMMigrationTask.java [hunk 3]
```diff
@@ -52,19 +52,21 @@ public void transform(NodeType node) {
...
         };
 
+        boolean modified = false;
+
         for (String componentName : componentsNameToAffect) {
             IComponentFilter componentFilter = new NameComponentFilter(componentName);
 
             try {
-                ModifyComponentsAction.searchAndModify(item, processType,
+                modified = modified || ModifyComponentsAction.searchAndModify(item, processType,
```
Comment (ozhelezniak-talend) [inline main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/EncryptClientSecretMicrosoftCRMMigrationTask.java:61]: || is a bad choice here.

Comment (ozhelezniak-talend) [reply main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/EncryptClientSecretMicrosoftCRMMigrationTask.java:61]: Also, I asked for a separate WorkItem. Don't mess with different changes for different tickets.

**Review by ozhelezniak-talend — CHANGES_REQUESTED** [review]

File: main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/RemoveDestinationPropertyExtractJSONFieldsMigrationTask.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                    nodeType.getElementParameter().remove(destination);
+                }
+            }
+        };
+
+        IComponentFilter componentFilter = new NameComponentFilter(componentNameToAffect);
+
+        try {
+            boolean modified = ModifyComponentsAction.searchAndModify(item, processType,
+                    componentFilter, Arrays.asList(removeDestinationProp));
```
Comment (ozhelezniak-talend) [inline main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/RemoveDestinationPropertyExtractJSONFieldsMigrationTask.java:48]: You could use here Collections.singletonList

**Review by ozhelezniak-talend — APPROVED** [review]
lgtm

**Pull request merged**